common/src/main/java/com/wynntils/functions/CharacterFunctions.java

```diff
@@ -101,44 +103,61 @@ public List<String> getAliases() {
     }
 
     public static class ClassFunction extends Function<String> {
-        // FIXME: original had upper/lower case versions. Make a upper/lower function instead.
         @Override
-        public String getValue(String argument) {
-            return Models.Character.getActualName();
+        public String getValue(FunctionArguments arguments) {
+            Boolean showReskinnedName =
+                    (Boolean) arguments.getArgument("showReskinnedName").getValue();
```

Comment:
I think we can take a page out of GSON's book here. If you add a `asBoolean()` to the argument class, we can get rid of the casting when using it. (It needs to check for type and throw an exception in case of type mismatch).

And probably you'll need `asString`, `asInt`, and `asDouble` as well...

Reply:
Sounds good, yes.